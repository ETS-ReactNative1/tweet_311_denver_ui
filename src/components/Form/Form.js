import React, { Component } from 'react';
import { View, StyleSheet, Text, TextInput, Image, TouchableOpacity, ScrollView, Switch } from 'react-native';

class Form extends Component {
  state = {
      email: '',
      description: '',
      isSnowRemoval: false,
      loaction: '',
      photo: '',
      error: ''
    }

  validateSubmit = () => {
    const { description } = this.state;
    return this.validateEmail() && description ? true : false;
  }

  validateEmail = () => {
    return /\S+@\S+\.\S+/.test(this.state.email);
  }

  handleSubmit = () => {
    const payload = {
      report: {
        category: this.state.isSnowRemoval ? 'snow_removal' : 'other',
        description: this.state.description,
        image: this.state.images,
        email: this.state.email
      },
      location: {
        lat: '',
        lon: ''
      }
    };
    if (this.validateSubmit()) {
      // this is where we make the API call
      // set App's decsription state
      // redirect to Tweet
    } else {
        this.setState({error: 'Please add an email, description, and location.'})
    }
  }

  render() {
    return (
      <View style={styles.container}>
        <ScrollView>
          <Text style={styles.h1}>
            Tweet<Text style={styles.h1Color}>311</Text>Denver
          </Text>
          <Text style={styles.label}>Enter Email:</Text>
          <TextInput style={styles.smallInput} value={this.state.email} onChangeText={text => this.setState({email: text})}/>
          <View style={styles.smallWrapper}>
            <Switch style={styles.CheckBox} value={this.state.isSnowRemoval} onValueChange={value => this.setState({isSnowRemoval: value})}/>
            <Text style={styles.iconLabel}>Snow Removal?</Text>
          </View>
          <Text style={styles.label}>Description:</Text>
          <TextInput multiline={true} style={styles.largeInput} value={this.state.description} onChangeText={text => this.setState({description: text})}/>
          <View style={styles.smallWrapper}>
            <Image style={styles.icon} source={require('../../../assets/images/placeholder.png')}/>
            <Text style={styles.iconLabel}>Add Location</Text>
          </View>
          <View style={styles.smallWrapper}>
            <Image style={styles.icon} source={require('../../../assets/images/photo.png')}/>
            <Text style={styles.iconLabel}>Add Photo</Text>
          </View>
          <Text style={styles.error}>{this.state.error}</Text>
          <TouchableOpacity style={styles.button}><Text style={styles.buttonLabel} onPress={() => this.handleSubmit()}>Submit</Text></TouchableOpacity>
        </ScrollView>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff'
  },
  h1: {
    color: '#3976EA',
    fontSize: 45,
    marginTop: 60,
    marginLeft: 'auto',
    marginRight: 'auto'
  },
  h1Color: {
    color: '#000000'
  },
  smallInput: {
    borderColor: '#000000',
    borderWidth: 1,
    fontSize: 30,
    height: 50,
    marginLeft: 'auto',
    marginRight: 'auto',
    padding: 5,
    width: '80%'
  },
  largeInput: {
    borderColor: '#000000',
    borderWidth: 1,
    fontSize: 30,
    height: 200,
    marginLeft: 'auto',
    marginRight: 'auto',
    padding: 5,
    width: '80%'
  },
  label: {
    color: '#3976EA',
    fontSize: 20,
    marginBottom: 10,
    marginLeft: '10%',
    marginTop: 40
  },
  smallWrapper: {
    alignItems: 'center',
    alignSelf: 'center',
    flex: 0,
    flexDirection: 'row',
    height: 50,
    marginLeft: 'auto',
    marginRight: 'auto',
    marginTop: 40,
    width: '80%'
  },
  icon: {
    height: 50,
    marginRight: 15,
    width: 50
  },
  iconLabel: {
    color: '#3976EA',
    fontSize: 25
  },
  button: {
    alignItems: 'center',
    backgroundColor: '#3976EA',
    borderRadius: 20,
    height: 40,
    justifyContent: 'center',
    marginBottom: 40,
    marginLeft: 'auto',
    marginRight: 'auto',
    marginTop: 40,
    width: '40%'
  },
  buttonLabel: {
    color: '#FFFFFF',
    fontSize: 20
  },
  CheckBox: {
    marginRight: 15
  },
  error: {
    color: 'red',
    marginLeft: 'auto',
    marginRight: 'auto',
    marginTop: 40,
    width: '80%'
  }
});

export default Form;
