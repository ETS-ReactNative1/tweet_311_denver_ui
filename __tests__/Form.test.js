import React from 'react';
import renderer from 'react-test-renderer';

import Form from '../src/components/Form/Form';

describe('Form', () => {

  test('renders Form correctly', () => {
    const mockLocation = { location: {
        lat: '1233',
        long: '-24324'
      }
    } 
    const tree = renderer.create(<Form location={mockLocation}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
