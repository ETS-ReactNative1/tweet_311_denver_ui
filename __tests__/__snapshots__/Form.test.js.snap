// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders Form correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <Text
        style={
          Object {
            "color": "#3976EA",
            "fontSize": 45,
            "marginLeft": "auto",
            "marginRight": "auto",
            "marginTop": 60,
          }
        }
      >
        Tweet
        <Text
          style={
            Object {
              "color": "#000000",
            }
          }
        >
          311
        </Text>
        Denver
      </Text>
      <Text
        style={
          Object {
            "color": "#3976EA",
            "fontSize": 20,
            "marginBottom": 10,
            "marginLeft": "10%",
            "marginTop": 40,
          }
        }
      >
        Enter Email:
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#000000",
            "borderWidth": 1,
            "fontSize": 30,
            "height": 50,
            "marginLeft": "auto",
            "marginRight": "auto",
            "padding": 5,
            "width": "80%",
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flex": 0,
            "flexDirection": "row",
            "height": 50,
            "marginLeft": "auto",
            "marginRight": "auto",
            "marginTop": 40,
            "width": "80%",
          }
        }
      >
        <RCTSwitch
          accessibilityRole="button"
          onChange={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "height": 31,
                "width": 51,
              },
              Object {
                "marginRight": 15,
              },
            ]
          }
          value={false}
        />
        <Text
          style={
            Object {
              "color": "#3976EA",
              "fontSize": 20,
            }
          }
        >
          Snow Removal?
        </Text>
      </View>
      <Text
        style={
          Object {
            "color": "#3976EA",
            "fontSize": 20,
            "marginBottom": 10,
            "marginLeft": "10%",
            "marginTop": 40,
          }
        }
      >
        Description:
      </Text>
      <TextInput
        allowFontScaling={true}
        multiline={true}
        onChangeText={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#000000",
            "borderWidth": 1,
            "fontSize": 20,
            "height": 200,
            "marginLeft": "auto",
            "marginRight": "auto",
            "padding": 5,
            "width": "80%",
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(0, 0, 0, 0.1)",
            "borderRadius": 4,
            "height": 40,
            "justifyContent": "center",
            "marginLeft": "auto",
            "marginRight": "auto",
            "marginTop": 40,
            "opacity": 1,
            "width": "80%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#3976EA",
              "fontSize": 20,
            }
          }
        >
          Add Location
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(0, 0, 0, 0.1)",
            "borderRadius": 4,
            "height": 40,
            "justifyContent": "center",
            "marginLeft": "auto",
            "marginRight": "auto",
            "marginTop": 40,
            "opacity": 1,
            "width": "80%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#3976EA",
              "fontSize": 20,
            }
          }
        >
          Add Photo
        </Text>
      </View>
      <Text
        style={
          Object {
            "color": "red",
            "marginLeft": "auto",
            "marginRight": "auto",
            "marginTop": 40,
            "width": "80%",
          }
        }
      >
        
      </Text>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#3976EA",
            "borderRadius": 20,
            "height": 40,
            "justifyContent": "center",
            "marginBottom": 40,
            "marginLeft": "auto",
            "marginRight": "auto",
            "marginTop": 40,
            "opacity": 1,
            "width": "40%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 20,
            }
          }
        >
          Submit
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
